test/integration/confirmations/transactions/contract-interaction.test.tsx

```diff
@@ -0,0 +1,428 @@
+        usdConversionRate: 3414.67,
+      },
+    },
+    currentCurrency: 'usd',
+    pendingApprovals: {
+      [pendingTransactionId]: {
+        id: pendingTransactionId,
+        origin: 'origin',
+        time: pendingTransactionTime,
+        type: ApprovalType.Transaction,
+        requestData: {
+          txId: pendingTransactionId,
+        },
+        requestState: null,
+        expectsResult: false,
+      },
+    },
+    pendingApprovalCount: 1,
+    knownMethodData: {
+      '0x3b4b1381': {
+        name: 'Mint NFTs',
+        params: [
+          {
+            type: 'uint256',
+          },
+        ],
+      },
+    },
+    transactions: [
+      getUnapprovedTransaction(
+        accountAddress,
+        pendingTransactionId,
+        pendingTransactionTime,
+      ),
+    ],
+  };
+};
+
+const advancedDetailsMockedRequests = {
+  getGasFeeTimeEstimate: {
+    lowerTimeBound: new Date().getTime(),
+    upperTimeBound: new Date().getTime(),
+  },
+  getNextNonce: '9',
+  decodeTransactionData: {
+    data: [
+      {
+        name: 'mintNFTs',
+        params: [
+          {
+            name: 'numberOfTokens',
+            type: 'uint256',
+            value: 1,
+          },
+        ],
+      },
+    ],
+    source: 'Sourcify',
+  },
+};
+
+const setupSubmitRequestToBackgroundMocks = (
+  mockRequests?: Record<string, unknown>,
+) => {
+  mockedBackgroundConnection.submitRequestToBackground.mockImplementation(
+    createMockImplementation({
+      ...advancedDetailsMockedRequests,
+      ...(mockRequests ?? {}),
+    }),
+  );
+};
+
+const getMetaMaskStateWithMaliciousUnapprovedContractInteraction = (
+  accountAddress: string,
+) => {
+  return {
+    ...getMetaMaskStateWithUnapprovedContractInteraction({ accountAddress }),
+    transactions: [
+      getMaliciousUnapprovedTransaction(
+        accountAddress,
+        pendingTransactionId,
+        pendingTransactionTime,
+      ),
+    ],
+  };
+};
+
+describe('Contract Interaction Confirmation', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    setupSubmitRequestToBackgroundMocks();
+    mock4byte();
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  it('displays the header account modal with correct data', async () => {
+    const account =
+      mockMetaMaskState.internalAccounts.accounts[
+        mockMetaMaskState.internalAccounts
+          .selectedAccount as keyof typeof mockMetaMaskState.internalAccounts.accounts
+      ];
+
+    const accountName = account.metadata.name;
+    const mockedMetaMaskState =
+      getMetaMaskStateWithUnapprovedContractInteraction({
+        accountAddress: account.address,
+      });
+
+    const { getByTestId, queryByTestId, findByTestId } =
+      await integrationTestRender({
+        preloadedState: mockedMetaMaskState,
+        backgroundConnection: backgroundConnectionMocked,
+      });
+
+    expect(getByTestId('header-account-name')).toHaveTextContent(accountName);
```

Comment:
Just curious is it possible to not add data-test-id and rather use other selectors.